lib/backends/cpu/op-resolve-rules.ts

```diff
@@ -0,0 +1,90 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+
+import {FLOAT_TYPES, NUMBER_TYPES} from '../../operators';
+import {OpSet} from '../../opset';
+
+import {CpuArgMax} from './ops/argMax';
+import {CpuBatchNormalization} from './ops/batch-normalization';
+import {CpuBinaryOp} from './ops/binary-op';
+import {CpuConcat} from './ops/concat';
+import {CpuConv} from './ops/conv';
+import {CpuDropout} from './ops/dropout';
+import {CpuFlatten} from './ops/flatten';
+import {CpuGather} from './ops/gather';
+import {CpuGemm} from './ops/gemm';
+import {CpuImageScaler} from './ops/image-scaler';
+import {CpuInstanceNormalization} from './ops/instance-normalization';
+import {CpuLrn} from './ops/lrn';
+import {CpuMatMul} from './ops/matmul';
+import {CpuAveragePool, CpuGlobalAveragePool, CpuGlobalMaxPool, CpuMaxPool} from './ops/pool';
+import * as cpuReduce from './ops/reduce';
+import {CpuReshape} from './ops/reshape';
+import {CpuSlice} from './ops/slice';
+import {CpuSoftmax} from './ops/softmax';
+import {CpuSqueeze} from './ops/squeeze';
+import {CpuSum} from './ops/sum';
+import {CpuTile} from './ops/tile';
+import {CpuTranspose} from './ops/transpose';
+import * as unaryOps from './ops/unary-op';
+import {CpuUnsqueeze} from './ops/unsqueeze';
+
+export const CPU_OP_RESOLVE_RULES: ReadonlyArray<OpSet.ResolveRule> = [
+  ['Abs', '', '6+', () => new unaryOps.CpuUnaryOp(NUMBER_TYPES, unaryOps.abs)],
+  ['Acos', '', '7+', () => new unaryOps.CpuUnaryOp(NUMBER_TYPES, unaryOps.acos)],
+  ['Add', '', '7+', () => new CpuBinaryOp(NUMBER_TYPES, (e1, e2) => (e1 + e2))],
+  ['And', '', '7+', () => new CpuBinaryOp(['bool'], (e1, e2) => (e1 && e2))],
+  ['ArgMax', '', '1+', () => new CpuArgMax()],
+  ['Asin', '', '7+', () => new unaryOps.CpuUnaryOp(NUMBER_TYPES, unaryOps.asin)],
+  ['Atan', '', '7+', () => new unaryOps.CpuUnaryOp(NUMBER_TYPES, unaryOps.atan)],
```

Comment:
How were the ResolveRules created for the very first time? How can we make sure it is accurate? Do we need to maintain any script(s) used to generate them ?

Reply:
this is a pain point. one possible way is to grab more test cases from other place.

meanwhile, I am writing a script to generate the operator support table from data (to replace /docs/operators.md). that helps but still not a guarantee of 100% accuracy.